### ISSUE TITLE
[2019 ICCV] Rethinking ImageNet Pre-training

### ISSUE THREAD

**Issue by Jasonlee1995** [Jasonlee1995/AI_Papers#77]
기존의 computer vision task에서는 ImageNet pre-trained 모델을 backbone으로 이용하여 fine-tune하는 방식으로 학습했음

해당 논문은 detection task에서 random initialization한 모델과 pre-train + fine-tune 모델을 비교하며 contribution은 다음과 같음

1. Architecture에 변화를 크게 주지 않고 target task에서 scratch로 학습해도 됨
2. Scratch로 학습하면 converge하는데 iteration이 더 많이 걸림
3. Data가 충분한 경우, scratch로 학습한 성능이 pre-train + fine-tune 성능과 비슷함
4. ImageNet pre-training은 convergence speed-up을 하는데 도움을 줌
5. ImageNet pre-training은 small data regime가 아닌 경우 overfitting을 막는다고 보기 어려움
6. ImageNet pre-training은 target task가 classification보다 localization에 sensitive할수록 영향력이 없어짐

## 1. Methodology

ImageNet에서 사용되는 모델들의 경우 대부분 Batch Normalization (BN)을 사용함

근데 object detection dataset의 image size가 크기에 small batch를 사용할 수 밖에 없음

Small batch size는 BN의 accuracy를 degrade하기에, fine-tuning에서는 BN을 freeze하여 학습했음

Scratch부터 학습할 경우 BN을 freeze하는 것이 불가능하므로, Group Normalization (GN) 혹은 Synchronized Batch Normalization (SyncBN)을 사용하여 해결

## 2. Experiments Settings

Mask R-CNN with ResNet or ResNeXt + Feature Pyramid Network (FPN) backbone

Scratch로 학습할 모델은 기존의 fine-tune의 lr과 scheduler를 따라가되, 감소된 learning rate iteration만 같게 유지
(fine-tune lr scheduler가 epoch 100에 60, 80이었다면 scratch는 300에 260, 280이라는 말)

First large learning rate를 오래 학습하는 것이 유용하며, small learning rate를 오래 학습하면 종종 overfitting된다고 함

COCO dataset의 경우 train 2017로 학습하고 val 2017로 evaluation

PASCAL VOC dataset의 경우 trainval 2007 + train 2012로 학습하고 val 2012로 evaluation

나머지 자세한 실험 세팅은 논문 참고

#### 3.1.1. Baselines
<img width="593" alt="image" src="https://user-images.githubusercontent.com/49643709/175625210-927637f8-3ef3-4029-88d7-4c776cab25eb.png">
<img width="589" alt="image" src="https://user-images.githubusercontent.com/49643709/175625643-0e49ec70-181b-46fc-829f-474813573ade.png">
<img width="589" alt="image" src="https://user-images.githubusercontent.com/49643709/175625946-26194d89-69de-4bb5-8136-d192aef40c1b.png">
<img width="580" alt="image" src="https://user-images.githubusercontent.com/49643709/175633895-08eb0099-1b70-455e-8e8c-5a477db5c49b.png">

Pre-train + fine-tune의 경우 적은 iteration으로 near optimum으로 converge

Scratch로 모델을 오래 학습하면 pre-train + fine-tune 성능을 따라잡을 수 있음

즉 ImageNet pre-trained model을 이용하여 fine-tune하는 것은 convergence speed를 빠르게 할 뿐, final detection accuracy에 대해 아무런 improvement는 없음

#### 3.1.2. Multiple detection metrics
<img width="592" alt="image" src="https://user-images.githubusercontent.com/49643709/175634178-b41d82da-275b-47e4-bfa1-a45515ab1e65.png">

Scratch로 학습한 모델이 다양한 AP metric에서 비슷한 결과를 보임

즉 scratch로 학습한 모델이 단순히 singel metric만 따라잡는 것이 아니라 전반적인 metric을 다 따라잡음

#### 3.1.3. Enhanced baselines
<img width="437" alt="image" src="https://user-images.githubusercontent.com/49643709/175641822-5f5a732b-f137-479f-8f59-71b427e940f8.png">

Detection performance를 높이는 다양한 테크닉을 이용했을 때에도 scratch로 학습한 모델이 pre-train + fine-tune 모델과 비슷한 성능을 가짐

#### 3.1.4. Large models trained from scratch
<img width="442" alt="image" src="https://user-images.githubusercontent.com/49643709/175643561-bb9be141-6286-4035-9a79-9897bbadf0f8.png">

위에서 진행한 실험에서는 ResNet 101 모델을 사용했는데 이보다 4배 정도 FLOPs가 더 많은 ResNeXt-152 8x32d 모델을 이용

큰 모델을 사용했음에도 불구하고 따로 overfitting 현상은 관측되지 않았으며 scatch 모델이 pre-train + fine-tune 모델과 비슷한 성능을 가짐

#### 3.1.5. Keypoint detection
<img width="434" alt="image" src="https://user-images.githubusercontent.com/49643709/175644474-c9aaa6a4-22db-4580-ac9d-26e5bb633e39.png">

Train iteration을 늘리지 않고 scratch 모델이 pre-train + fine-tune 모델과 비슷한 성능을 가짐

이가 의미하는 바는 ImageNet pre-training은 localization information이 적기에, keypoint detection에선 도움이 안된다는 것을 알 수 있음

#### 3.2.1. 35k, 10k COCO training images
<img width="882" alt="image" src="https://user-images.githubusercontent.com/49643709/175644820-f907b272-fa2a-44df-a805-88a345950487.png">

35k COCO training images : COCO train dataset의 약 1/3

10k COCO training images : COCO train dataset의 약 1/10

두 경우 모두 scratch로 학습한 모델이 pre-train + fine-tune 모델과 비슷한 성능을 가짐

#### 3.2.2. Breakdown regime: 3.5k, 1k COCO training images
<img width="440" alt="image" src="https://user-images.githubusercontent.com/49643709/175645152-039bca63-cf33-4561-ba3a-2f1c4d00c335.png">

7.5k COCO training images : COCO train dataset의 약 1/30

1k COCO training images : COCO train dataset의 약 1/100

두 경우 모두 scratch로 학습한 모델이 loss는 pre-train + fine-tune 모델과 비슷하게 converge하나 성능은 안좋음

즉 data의 부족으로 인한 overfitting이 일어남

Scratch 모델과 pre-train + fine-tune 모델의 성능 차이가 나게 되는 beakdown point는 COCO dataset 기준으로 3.5k ~ 10k 사이에 있음

#### 3.2.3. Breakdown regime: PASCAL VOC

Faster R-CNN with ResNet 101 + FPN + FN + training-time augmentation으로 실험

Scratch 모델의 성능이 pre-train + fine-tune 모델보다 낮음

저자들은 반대되는 현상이 일어나는 이유에 대해 VOC dataset의 경우 instance와 category가 적은 것이 training data가 부족한 현상과 비슷한 impact를 준다고 제시함

## 4. Discussions

저자들은 해당 관측에 기반하여 몇몇 질문들에 대한 답을 제시함

### 4.1. Is ImageNet pre-training necessary? No

ImageNet은 convergence speed-up에 도움이 되나 target data가 충분한 경우 반드시 필요하지 않음

Target data의 annotation을 collect하는게 오히려 성능 향상에 영향을 줄 것임

### 4.2. Is ImageNet helpful? Yes

ImageNet pre-training은 research cycle을 줄여주고 결과를 빨리 확인하는데 도움을 줌

이는 computer vision research를 하는데 있어서 매우 중요한 이점임

### 4.3. Do we need big data? Yes

Classification-level pre-training을 수집하고 정제하는 것보다 target domain의 data를 collect하는 것이 더 효과적임

### 4.4. Shall we pursuit universal representations? Yes

Learning universal representations is a laudable goal
